### PR TITLE
[trivial] Add doxygen-compatible comments to CAffectedKeysVisitor

### PR DESCRIPTION
**Pull request by PierreRochard**

### CODE REVIEW

File: src/wallet/wallet.cpp [hunk 1]
```diff
@@ -102,14 +102,26 @@ std::string COutput::ToString() const
...
+/** A class to identify which pubkeys a script and a keystore have in common. */
 class CAffectedKeysVisitor : public boost::static_visitor<void> {
 private:
     const CKeyStore &keystore;
     std::vector<CKeyID> &vKeys;
 
 public:
+    /**
+     * @param keystoreIn The CKeyStore that is queried for the presence of a pubkey.
+     * @param vKeysIn A vector to which a script's pubkey identifiers are appended if they are in the keystore.
```
Comment (Empact) [inline src/wallet/wallet.cpp]: nit: could declare these [in]

File: src/wallet/wallet.cpp [hunk 2]
```diff
@@ -102,14 +102,26 @@ std::string COutput::ToString() const
...
+     */
     CAffectedKeysVisitor(const CKeyStore &keystoreIn, std::vector<CKeyID> &vKeysIn) : keystore(keystoreIn), vKeys(vKeysIn) {}
 
+    /** 
+     * Apply the visitor to each destination in a script, recursively to the redeemscript
+     * in the case of p2sh destinations.
+     * @return void
```
Comment (Empact) [inline src/wallet/wallet.cpp]: nit: seems extraneous

File: src/wallet/wallet.cpp [hunk 3]
```diff
@@ -102,14 +102,26 @@ std::string COutput::ToString() const
...
+     * @param script The CScript from which destinations are extracted.
+     * @post Any CKeyIDs that script and keystore have in common are appended to vKeys.
```
Comment (Empact) [inline src/wallet/wallet.cpp]: nit: maybe make it explicit that they’re added to the visitor’s vKeys

**Review by Empact — COMMENTED** [review]
ACK would improve pages like this one: https://dev.visucore.com/bitcoin/doxygen/class_c_affected_keys_visitor.html

File: src/wallet/wallet.cpp [hunk 4]
```diff
@@ -102,14 +102,26 @@ std::string COutput::ToString() const
...
     std::vector<CKeyID> &vKeys;
 
 public:
+    /**
+     * @param keystoreIn The CKeyStore that is queried for the presence of a pubkey.
+     * @param vKeysIn A vector to which a script's pubkey identifiers are appended if they are in the keystore.
+     */
     CAffectedKeysVisitor(const CKeyStore &keystoreIn, std::vector<CKeyID> &vKeysIn) : keystore(keystoreIn), vKeys(vKeysIn) {}
 
+    /**
```
Comment (sipa) [inline src/wallet/wallet.cpp]: Linter complains about trailing whitespace here.

Comment (PierreRochard) [reply src/wallet/wallet.cpp]: Thanks, fixed

File: src/wallet/wallet.cpp (hunk 1 above)
Comment (Empact) [reply src/wallet/wallet.cpp]: https://www.stack.nl/~dimitri/doxygen/manual/commands.html#cmdparam

Comment (PierreRochard) [reply src/wallet/wallet.cpp]: Thanks, fixed

**Comment by Empact** [comment]
ACK 3339d84

**Comment by fanquake** [comment]
utACK 3339d84

**Comment by laanwj** [comment]
thanks for adding developer documentation!

utACK 3339d845354c9c357ec90505192748d9d639e72e

**Pull request merged**